coreUtilities/src/main/java/org/opensearch/migrations/transformation/entity/Entity.java

```diff
@@ -0,0 +1,19 @@
+package org.opensearch.migrations.transformation.entity;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Common interface for entities that are used to perform transformations
+ *
+ * Notes on performance: ObjectNode requires a fully marshalled json, for
+ * less than 1mb json bodies this will likely be fine; however, if dealing with larger
+ * objects or perf issues are encountered should consider adding an alternative
+ * path for JsonParser.  This might be needed for processing document bodies
+ * specifically.
```

Comment:
This is an interesting comment.  Do you think that we might setup some transformations that are single-pass, where we wouldn't realize the entire DOM-tree in memory?  As it is, the replayer currently uses the JsonParser to at least stream the results in and the JsonEmitter to stream them out, to avoid truly egregious buffering.  I don't know of a streaming transformation specification for json & fear that forcing users/devs to figure that out could be really challenging (I have experience here w/ SAX models for XML & it would be challenging for me).